tests/all/js_globals/Generator.rs

```diff
@@ -0,0 +1,51 @@
+#![allow(non_snake_case)]
+
+use project;
+
+#[test]
+fn next() {
+    project()
+        .file(
+            "src/lib.rs",
+            r#"
+            #![feature(proc_macro, wasm_custom_section)]
+
+            extern crate wasm_bindgen;
+            use wasm_bindgen::prelude::*;
+            use wasm_bindgen::js;
+
+            #[wasm_bindgen]
+            pub fn next(this: &js::Generator, value: &JsValue) -> JsValue {
+                this.next(value)
+            }
+        "#,
+        )
+        .file(
+            "test.ts",
+            r#"
+            import * as assert from "assert";
+            import * as wasm from "./out";
+
+            export function test() {
+                function* generator() {
+                    const reply = yield '2 * 2';
+
+                    return reply === 4;
+                }
+
+                const gen = generator();
```

Comment:
And then we should also have a test with:

```js
function* G() {
    throw new Error("uh oh");
}

assert(wasm.next_throws_error(G()));
```

where we have this rust:

```rust
#[wasm_bindgen]
pub fn next_throws_error(g: &Generator) -> bool {
    g.next().is_err()
}
```

Reply:
Rare use case but its real, you are right. Well thought!